### PR TITLE
remove AzureManagedCluster, add migration script

### LINKED ISSUE #1396: AzureManagedCluster is useless; let's remove it (and keep AzureManagedControlPlane)
/kind cleanup

**Describe the solution you'd like**
Eliminate AzureManagedCluster. It has absolutely no useful spec or status. We should simply point CAPI Clusters to AzureManagedControlPlane as the InfrastructureRef.

### PR DESCRIPTION
**Pull request by alexeldeib**
**What type of PR is this?**

/kind api-change
/kind cleanup

**What this PR does / why we need it**:
remove AzureManagedCluster. Provides a Go script to migrate Clusters using AMC to AMCP for the infra ref, and then delete the CRD (which also deletes all AMC instances).

There are no webhooks, so that should be sufficient? I'm not going to muck with RBAC, since it's all in a giant role right now. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1396

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
AzureManagedCluster has been removed in favor of AzureManagedControlPlane. AzureManagedControlPlane is now both the InfrastructureRef and the ControlPlaneRef for CAPI clusters. This is a BREAKING CHANGE. Action required: a migration script has been provided in `exp/cmd/migrate_managed_cluster`. Use it to migrate your existing CAPI cluster safely, and clean up leftover Kubernetes objects. It will set `spec.infrastructureRef = spec.controlPlaneRef` for all CAPI clusters using AzureManagedCluster as the infrastructure reference.
```

### CODE REVIEW

File: exp/cmd/migrate_managed_cluster/migrate_managed_cluster.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
+package main
```
Comment (alexeldeib) [inline exp/cmd/migrate_managed_cluster/migrate_managed_cluster.go:1]: no idea how to package this or where to put it. but I thought it's a nice touch!

**Review by CecileRobertMichon — COMMENTED** [review]
@alexeldeib given this is breaking, is the idea to get it in before v0.5.0?

**Comment by alexeldeib** [comment]
either way works for me. remind me, what's the target release date for v0.5.0?

**Comment by CecileRobertMichon** [comment]
> either way works for me. remind me, what's the target release date for v0.5.0?

we're about to cut CAPI v0.4.0 (by EOD), in this morning's office hours we discussed cutting CAPZ v0.5.0 by end of next week

**Comment by alexeldeib** [comment]
it'd be nice to make it but i'm not stressed if it doesn't. I'm mostly trying to get the implementation to a stable place with best practices covered, which really requires most of what I mentioned https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1243#issuecomment-864380359 more than this specific piece. I'd be (very) pleasantly surprised if we managed to finish all of that work in the next ~day, so definitely not married to the release cycle.

**Comment by devigned** [comment]
@alexeldeib what do you think we should do with this PR?

**Comment by CecileRobertMichon** [comment]
/close

based on https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1396#issuecomment-1055737796

**Pull request closed without merging**